drpv4/resource_drp_reservation.go

```diff
@@ -0,0 +1,249 @@
+package drpv4
+
+import (
+	"log"
+	"net"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"gitlab.com/rackn/provision/v4/models"
+)
+
+// resourceReservation represents a reservation in DRP system
+//
+//		resource "drp_reservation" "reservation" {
+//		  name = "reservation"
+//		  description = "reservation"
+//		  documenation = "reservation"
+//		  addr = "192.168.0.1"
+//	   duration = 86400
+//	   next_server = "192.168.2.1"
+//	   scoped = false
+//	   strategy = "strategy"
+//	   subnet = "subnet"
+//	   token = "token"
+//
+//	   options {
+//	     code = "value1"
+//	     value = "value2"
+//	   }
+//
+//	   options {
+//	     code = "value3"
+//	     value = "value4"
+//	   }
+//	}
+func resourceReservation() *schema.Resource {
+	r := &schema.Resource{
+		Create: resourceReservationCreate,
+		Read:   resourceReservationRead,
+		Update: resourceReservationUpdate,
+		Delete: resourceReservationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Reservation description",
+				Optional:    true,
+			},
+			"documentation": {
+				Type:        schema.TypeString,
+				Description: "Reservation documentation",
+				Optional:    true,
+			},
+			"address": {
+				Type:        schema.TypeString,
+				Description: "Reservation address",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"duration": {
+				Type:        schema.TypeInt,
+				Description: "Reservation duration",
+				Optional:    true,
+			},
+			"next_server": {
+				Type:        schema.TypeString,
+				Description: "Reservation next server",
+				Optional:    true,
+			},
+			"scoped": {
+				Type:        schema.TypeBool,
+				Description: "Reservation scoped",
+				Optional:    true,
+			},
+			"strategy": {
+				Type:        schema.TypeString,
+				Description: "Reservation strategy",
+				Optional:    true,
+				Default:     "MAC",
+			},
+			"subnet": {
+				Type:        schema.TypeString,
+				Description: "Reservation subnet",
+				Optional:    true,
+			},
+			"token": {
+				Type:        schema.TypeString,
+				Description: "Reservation token",
+				Required:    true,
```

Comment:
Shouldn't this be marked as sensitive?

Reply:
It doesn't need to be sensitive as it is not treated as sensitive by DRP.